python/device/device.py

```diff
@@ -1062,6 +1103,98 @@ def subtest_desired_properties(self):
             )
             self.metrics.get_twin_count_timed_out_out.increment()
 
+    def handle_method_thread(self, worker_thread_info):
+        while not self.done.isSet():
+            worker_thread_info.watchdog_epochtime = time.time()
+            if self.is_paused():
+                time.sleep(1)
+                continue
+
+            elif not self.is_pairing_complete():
+                time.sleep(1)
+                continue
+
+            # TODO: remove timeout or switch to handlers
+            logger.info("calling receive method function")
+            method_request = self.client.receive_method_request(timeout=10)
+            if method_request:
+                logger.info("Received method request {}".format(method_request.name))
+                if method_request.name == MethodNames.FAIL_WITH_404:
+                    logger.info("sending response to fail with 404")
+                    self.client.send_method_response(
+                        MethodResponse.create_from_method_request(method_request, 404)
+                    )
+                elif method_request.name == MethodNames.ECHO_REQUEST:
+                    self.client.send_method_response(
```

Comment:
There's no log here, but you log when we send a 404 or 500 response. Not sure if you only wanted to log the non-success HTTP status codes or if this was just a mistake

Reply:
Good point.